src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs

```diff
@@ -56,8 +56,8 @@ protected override void Dispose(bool disposing)
                 try
                 {
                     this.token.Cancel();
-                    this.token.Dispose();
                     this.exportTask.Wait();
+                    this.token.Dispose();
```

Comment:
Dispose `this.exportTask` after `this.exportTask.Wait()`?

Reply:
This is interesting. I found this article by Stephen Toub on why we don't have to dispose tasks in most cases: https://devblogs.microsoft.com/pfxteam/do-i-need-to-dispose-of-tasks/

Reply:
I think:
1. disposing the task is not the goal of this PR.
2. we need to refactor this class to replace task with explicit thread, there is already a comment https://github.com/open-telemetry/opentelemetry-dotnet/blob/bec00a9a928c296c04f9e0e65f26c2925441b575/src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs#L33.